src/Multicall3.sol

```diff
@@ -78,34 +111,71 @@ contract Multicall3 {
         }
     }
 
+    /// @notice Aggregate calls with a msg value
+    /// @notice Reverts if msg.value is less than the sum of the call values
+    /// @param calls An array of Call3Value structs
+    /// @return returnData An array of Result structs
+    function aggregate3Value(Call3Value[] calldata calls) public payable returns (Result[] memory returnData) {
```

Comment:
Could move this discussion to the other PR if we want, but might be nice to have the return values be consistent for `aggregate3` and `aggregate3Value`. But consistent with just `Result` or also with block info?

Reply:
IMO for on-chain calls, gas is most important, so we can just get the block number and hash by rpc call offchain.

Reply:
Solid logic 👌